tests/test_photon_ops.py

```diff
@@ -26,31 +22,210 @@ def test_lsst_optics() -> None:
+
+    # Compute a histogram with non uniform bins
+    # (spike regions and in-between spike regions):
+    h, _ = np.histogram(
+        delta_angles,
+        bins=sum(
+            (
+                (
+                    -spike_angle_tolerance / 2 + i * np.pi / 2.0,
+                    spike_angle_tolerance / 2 + i * np.pi / 2.0,
+                )
+                for i in range(5)
+            ),
+            (),
         ),
     )
+    # Merge last and first bin:
+    h[0] += h[-1]
+    h = h[:-1]
+
+    # Check that there are photons only in the spike regions:
+    np.testing.assert_array_equal(h[1::2], 0)
+    # Check that there are photons in all spike regions:
+    np.testing.assert_array_less(0, h[0::2])
+
+
+def extract_spike_angles(photon_array, x_center, y_center, r):
+    """Filters out a disc centered at (x_center, y_center) with radius r.
+    The reminding photons will be considered as spike photons.
+    Returns the angles of the spike photons wrt (x_center, y_center).
+    """
+    spike_photons = (
+        np.hypot(
+            photon_array.x - x_center,
+            photon_array.y - y_center,
+        )
+        > r
+    )
+    return np.arctan2(
+        photon_array.y[spike_photons] - y_center,
+        photon_array.x[spike_photons] - x_center,
+    )
+
+
+def test_lsst_diffraction_shows_field_rotation() -> None:
+    """Checks that the spikes rotate."""
+    lsst_diffraction = create_test_lsst_diffraction()
+    dt = 1.0
+    photon_array_0 = create_test_photon_array(t=0.0)
+    photon_array_1 = create_test_photon_array(t=dt)
+    local_wcs = create_test_wcs()
+    lsst_optics = create_test_lsst_optics()
+    lsst_diffraction.applyTo(photon_array_0, local_wcs=local_wcs)
+    lsst_optics.applyTo(photon_array_0, local_wcs=local_wcs)
+    lsst_diffraction.applyTo(photon_array_1, local_wcs=local_wcs)
+    lsst_optics.applyTo(photon_array_1, local_wcs=local_wcs)
+
+    # The expected image is contained in a disc + spikes outside the disc:
+    spike_angles_0 = extract_spike_angles(
+        photon_array_0,
+        x_center=564.5,
+        y_center=-1431.4,
+        r=20.0,
+    )
+    spike_angles_1 = extract_spike_angles(
+        photon_array_1,
+        x_center=564.5,
+        y_center=-1431.4,
+        r=20.0,
+    )
+
+    # Find the angle, the cross is rotated relative to the axis cross:
+    cross_rot_angle_0 = np.mean(spike_angles_0 % (np.pi / 2.0))
+    cross_rot_angle_1 = np.mean(spike_angles_1 % (np.pi / 2.0))
+
+    # Check that the angle of the crosses are rotated relative to each other:
+    expected_angle_difference = dt * (
+        diffraction.OMEGA_EARTH
+        * np.cos(lsst_diffraction.latitude / 180.0 * np.pi)
+        * np.cos(lsst_diffraction.azimuth / 180.0 * np.pi)
+        / np.cos(lsst_diffraction.altitude / 180.0 * np.pi)
+    )
+
+    np.testing.assert_almost_equal(
+        cross_rot_angle_1 - cross_rot_angle_0, expected_angle_difference, decimal=2
```

Comment:
decimal=2 kind of hides the level of precision here.  Should use `assert_allclose` and `rtol` to be clearer about the level of agreement.  I think this isn't currently very close.  I'm getting 0.018 vs 0.026, which is pretty different.  Do you understand where the discrepancy is coming from?

Reply:
I first thought, that the field rotation rate is changing significantly this close to zenith. But I now had a look and after 1sec,
we only have a change in the rate of `-3.600097e-05`. This cannot explain the difference. Have to investigate.

Reply:
According to `field_rotation_matrix`, I would expect a value of `0.02618501` rad at `t=1` and this position. At least so far this seems to agree.
The only place in the code this angle kicks in is at
```py
    shift = rot(
        diffraction_delta(rot_inv(pos), -v[:, 2], wavelength, geometry, distribution)
    )
```
So question (@rmjarvis / @jmeyers314 ):
is one of `rot`, `rot_inv` actually not needed? Is maybe the rotation of `v` already taken care of in the photon array?

Reply:
I think, I just messed up the order of arguments in `star_trace`.
Now:
```
ACTUAL: 0.022242493576406397
DESIRED: 0.025522087833554367
```